pkg/controllers/nodeclaim/notready/controller.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notready
+
+import (
+	"context"
+	"time"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	nodeclaimutil "sigs.k8s.io/karpenter/pkg/utils/nodeclaim"
+
+	corev1 "k8s.io/api/core/v1"
+
+	v1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+)
+
+// Controller NotReady is a nodeclaim controller that deletes nodeclaims when they have been unreachable for too long
+type Controller struct {
+	kubeClient client.Client
+}
+
+// NewController constructs a nodeclaim disruption controller
+func NewController(kubeClient client.Client) *Controller {
+	return &Controller{
+		kubeClient: kubeClient,
+	}
+}
+
+func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
+	if nodeClaim.Spec.UnreachableTimeout.Duration == nil {
+		return reconcile.Result{}, nil
+	}
+
+	node, err := nodeclaimutil.NodeForNodeClaim(ctx, c.kubeClient, nodeClaim)
+	if err != nil {
+		return reconcile.Result{}, nodeclaimutil.IgnoreDuplicateNodeError(nodeclaimutil.IgnoreNodeNotFoundError(err))
+	}
+
+	for _, taint := range node.Spec.Taints {
+		if taint.Key == corev1.TaintNodeUnreachable {
+			if taint.TimeAdded != nil {
+				durationSinceTaint := time.Since(taint.TimeAdded.Time)
+				if durationSinceTaint > *nodeClaim.Spec.UnreachableTimeout.Duration {
+					// if node is unreachable for too long, delete the nodeclaim
+					if err := c.kubeClient.Delete(ctx, nodeClaim); err != nil {
+						log.FromContext(ctx).V(0).Error(err, "Failed to delete nodeclaim", "node", node.Name)
```

Comment:
(nit)
```suggestion
						log.FromContext(ctx).V(0).Error(err, "Failed to delete NodeClaim", "node", node.Name)
```
?

And other places in the file.
At HTTP level, the lower case `nodeclaims` name is plural. The API kind NodeClaim uses UpperCamelCase.

Reply:
Thank you, adapted the UpperCamelCase NodeClaim in all logs